system/jlib/jflz.cpp

```diff
@@ -723,11 +732,28 @@ class jlib_decl CFastLZCompressor : public CInterface, public ICompressor
 
     size32_t write(const void *buf,size32_t len)
     {
+        // fill internal buffer with smaller chunks for better fit/less gap fill
+        size32_t lenb = wrmax;
+        if (len > lenb)
+        {
+            byte *b = (byte *)buf;
+            size32_t lentot = 0;
+            while (lentot < len)
+            {
+                if ((lentot + lenb) > len)
+                    lenb = len - lentot;
+                size32_t amnt = write((const void *)(b+lentot), lenb);
```

Comment:
I have re-written it to be a loop similar to above, trying to write up to wrmax (== max at open) bytes.  This also works well to keep small gaps when writing blocks out.